src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -586,17 +598,57 @@ public void onNewWorkingCopy(NewWorkingCopyEvent event)
          }
       });
       
-      events.addHandler(PopoutDocEvent.TYPE, this);
-      events.addHandler(DocWindowChangedEvent.TYPE, this);
-      events.addHandler(DocTabDragInitiatedEvent.TYPE, this);
-      events.addHandler(PopoutDocInitiatedEvent.TYPE, this);
-      events.addHandler(DebugModeChangedEvent.TYPE, this);
-      events.addHandler(ReplaceRangesEvent.TYPE, this);
-      events.addHandler(GetEditorContextEvent.TYPE, this);
-      events.addHandler(SetSelectionRangesEvent.TYPE, this);
-      events.addHandler(OpenProfileEvent.TYPE, this);
-      events.addHandler(RequestDocumentSaveEvent.TYPE, this);
-      events.addHandler(RequestDocumentCloseEvent.TYPE, this);
+      events_.addHandler(PopoutDocEvent.TYPE, this);
+      events_.addHandler(DocWindowChangedEvent.TYPE, this);
+      events_.addHandler(DocTabDragInitiatedEvent.TYPE, this);
+      events_.addHandler(PopoutDocInitiatedEvent.TYPE, this);
+      events_.addHandler(DebugModeChangedEvent.TYPE, this);
+      events_.addHandler(ReplaceRangesEvent.TYPE, this);
+      events_.addHandler(GetEditorContextEvent.TYPE, this);
+      events_.addHandler(SetSelectionRangesEvent.TYPE, this);
+      events_.addHandler(OpenProfileEvent.TYPE, this);
+      events_.addHandler(RequestDocumentSaveEvent.TYPE, this);
+      events_.addHandler(RequestDocumentCloseEvent.TYPE, this);
+      
+      initialized_ = true;
+   }
+
+   public void loadFullSource()
+   {
+      AceEditor.preload();
+
+      // sync UI prefs with shortcut manager
+      userPrefs_ = RStudioGinjector.INSTANCE.getUserPrefs();
+
+      if (userPrefs_.editorKeybindings().getValue() == UserPrefs.EDITOR_KEYBINDINGS_VIM)
+         ShortcutManager.INSTANCE.setEditorMode(KeyboardShortcut.MODE_VIM);
+      else if (userPrefs_.editorKeybindings().getValue() == UserPrefs.EDITOR_KEYBINDINGS_EMACS)
+         ShortcutManager.INSTANCE.setEditorMode(KeyboardShortcut.MODE_EMACS);
+      else if (userPrefs_.editorKeybindings().getValue() == UserPrefs.EDITOR_KEYBINDINGS_SUBLIME)
+         ShortcutManager.INSTANCE.setEditorMode(KeyboardShortcut.MODE_SUBLIME);
+      else
+         ShortcutManager.INSTANCE.setEditorMode(KeyboardShortcut.MODE_DEFAULT);
+   
+      events_.fireEvent(new DocTabsChangedEvent(null,
```

Comment:
Can we document the intent here? (why do we need to fire this "empty" DocTabsChangedEvent?)